src/common/container.h

```diff
@@ -93,6 +93,24 @@ static inline void smartlist_swap(smartlist_t *sl, int idx1, int idx2)
   }
 }
 
+/** Return true if there is shallow equality between smartlists -
+ * i.e. all indices correspond to exactly same object (pointer
+ * values are matching). Otherwise, return false.
+ */
+static inline int smartlist_shallow_eq(const smartlist_t *s1,
```

Comment:
Optional suggestion: We can make this function return `bool` instead of `int`.

Reply:
We generally don't use `bool` in the codebase, so I'll rather use `int`, like we do pretty much always.

Reply:
stdbool was only recently added to the codebase that's why it's not yet used extensively. In any case, I'm fine with int as well.